frontend/src/pages/lesson.js

```diff
@@ -1,22 +1,30 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
 import Container from "react-bootstrap/Container";
+import { useParams } from "react-router-dom";
 import QuizPanel from "../components/quizPanel";
+import { LessonContext } from "../utils/lessonContext";
 
 const LessonQuiz = () => {
-  const filler = {
-    word: "じゃね",
-    choice_0: "goodbye",
-    choice_1: "thank you",
-    choice_2: "maybe",
-    choice_3: "you",
-  };
-  return (
+  const { id } = useParams();
+  const { lesson, setLessonId } = useContext(LessonContext);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLessonId(id);
+  });
+
+  useEffect(() => {
+    if (lesson) setLoading(false);
+  }, [lesson]);
+  return !loading && !lesson.completed ? (
```

Comment:
i think you dont need the loading here since you already implemented it in the context level